peps/pep-0767.rst

```diff
@@ -0,0 +1,550 @@
+PEP: 767
+Title: Read-only class and protocol attributes
+Author: <REQUIRED: list of authors' real names and optionally, email addrs>
+Sponsor: Carl Meyer <carl@oddbird.net>
+Discussions-To: <REQUIRED: URL of current canonical discussion thread>
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 18-Nov-2024
+Python-Version: 3.14
+
+
+Abstract
+========
+
+:pep:`705` introduced the :external+py3.13:data:`typing.ReadOnly` type qualifier
+to allow defining read-only :class:`typing.TypedDict` items.
+
+This PEP proposes expanding the scope of ``ReadOnly`` to class and protocol
+attributes, as a single concise way to mark them read-only. Some parity changes
+are also made to :data:`typing.Final`.
+
+Akin to the :pep:`705`, it makes no changes to setting attributes at runtime. Correct
+usage of read-only attributes is intended to be enforced only by static type checkers.
+
+
+Motivation
+==========
+
+The Python type system lacks a single concise way to mark an :term:`attribute` read-only.
+This feature is present in other statically and gradually typed languages
```

Comment:
I've changed "object-oriented" to "statically and gradually typed", hopefully it's not a misnomer